### PR TITLE
Add (partial) descriptions of undocumented flags

### PR DESCRIPTION
**Pull request by ajweeks**
This PR isn't intended to be merged, but instead as a starting point for a fix for the scripting reference.

All flags found in `allowed_flags` in `api.lua` that were missing have been added to the scripting reference in this PR, however because I don't know the exact purpose of most I've left them as ???.

The scripting reference is extremely useful; in this case had I been able to search for "JMC" my life would have been a lot easier, and I wouldn't have needed to clone and dig into the GENie source 🙂

### CODE REVIEW

**Comment by bkaradzic** [comment]
Yeah some of those are obsolete...
Feel free to add docs for `_Cpp<standard version>_`.

**Comment by ajweeks** [comment]
Which ones are obsolete? I'm not sure what the rest of the features are for so I can't give meaningful descriptions.

**Comment by bkaradzic** [comment]
`ATL`, others too. Just not sure who/when added them... @rhoot Any ideas?

**Comment by rhoot** [comment]
Went spelunking. For these flags:

`AntBuildDebuggable` was added in #321. Looks to make Visual Studio projects targeting Tegra debuggable.

`DebugEnvsDontMerge` and `DebugEnvsInherit` were added as `EnvironmentArgsDontMerge` and `EnvironmentArgsInherit` in premake pre-fork. Specifically in [80eff1c](https://github.com/premake/premake-core/commit/80eff1c9fdb9a184c88ee25b3e2c7b8c72c5a870), and then renamed in [7185aaf](https://github.com/premake/premake-core/commit/7185aaf0a0f90c23cda0a4d91633d61acdb43991). These affect Visual Studio debugger properties. It looks like the default when setting `debugenvs` (which is also not documented...) is for it to replace all environment variables. `DebugEnvsInherit` causes it to also inherit debug environments from vsprops files, toolchains, etc. The same as going to VS project properties -> Debugging -> Environment -> Edit, and unchecking "Inherit from parent or project defaults". `DebugEnvsDontMerge` causes the environment variables to not merge with the ones inherited from the parent process. The same as going to VS project properties -> Debugging -> and setting Merge Environment to No.

`DeploymentContent` was added in #139, and looks to be adding files to the project for inclusion in  deployment.

**Comment by ajweeks** [comment]
Thanks @rhoot, how does that look?

**Comment by bkaradzic** [comment]
Removed ATL and MFC: https://github.com/bkaradzic/GENie/commit/8335df2e78ae15470af9ace3214ba3860cdfa612

File: docs/scripting-reference.md [hunk 1]
```diff
@@ -630,16 +630,16 @@ Specifies build flags to modify the compiling or linking process. Multiple calls
 #### Arguments
 _flags_ - List of flag names from list below. Names are case-insensitive and ignored if not supported on a platform.
 
-* _AntBuildDebuggable_ - ???
-* _ATL_ - ???
+* _AntBuildDebuggable_ - Enables Visual Studio projects to target Android via the NVIDIA Nsight Tegra tools. ([#321](https://github.com/bkaradzic/GENie/pull/321))
```
Comment (rhoot) [inline docs/scripting-reference.md:633]: You can target the platform without that flag, you just can't debug without it. Maybe change the wording slightly to reflect that.

**Comment by ajweeks** [comment]
Alright, I've updated to latest master, removed obsolete flags, and resolved comments. @rhoot @bkaradzic

**Pull request merged**